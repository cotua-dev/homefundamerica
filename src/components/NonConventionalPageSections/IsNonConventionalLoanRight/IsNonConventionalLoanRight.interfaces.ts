export interface FetchIsNonConventionalLoanRightImageQuery {
    isNonConventionalLoanRightImage: {
        publicURL: string;
    };
}
