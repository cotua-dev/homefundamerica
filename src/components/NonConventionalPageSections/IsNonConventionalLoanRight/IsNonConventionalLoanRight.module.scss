section#isNonConventionalLoanRightSection {
    height: 60vh;

    @media screen and (max-width: 950px) {
        height: 100%;
        flex-direction: column-reverse;
    }
}
