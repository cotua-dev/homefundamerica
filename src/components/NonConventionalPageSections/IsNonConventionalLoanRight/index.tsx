import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { moveToStepper } from '../../../utils/moveToStepper';
import { FetchIsNonConventionalLoanRightImageQuery } from './IsNonConventionalLoanRight.interfaces';
import { isNonConventionalLoanRightSection } from './IsNonConventionalLoanRight.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';

export const IsNonConventionalLoanRight: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchIsNonConventionalLoanRightImageQuery>(graphql`
        query {
            isNonConventionalLoanRightImage: file(name: {eq: "AdobeStock_325107268"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section id={isNonConventionalLoanRightSection} className={sharedStyles.loanNegative}>
            <div className={sharedStyles.contentWrapper}>
                <h2>{`Is a Non-Conventional Loan Right For You?`}</h2>
                <strong>
                    {`Non-conventional loans may be the right solution when looking for:`}
                </strong>
                <ul>
                    <li>{`Expanded guidelines and enhanced options`}</li>
                    <li>{`Tailored mortgage solutions for unique financial situations`}</li>
                    <li>{`Customized Portfolio Loans and Interest-Only Mortgages`}</li>
                </ul>
                <button
                    className={sharedStyles.applyButton}
                    type="button"
                    onClick={moveToStepper}
                >Apply Now</button>
            </div>
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.isNonConventionalLoanRightImage.publicURL}
                    alt="Is a non conventional loan right image"
                />
            </div>
        </section>
    );
}
