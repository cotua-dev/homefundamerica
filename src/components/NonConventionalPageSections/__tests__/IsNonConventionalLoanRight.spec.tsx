import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { IsNonConventionalLoanRight } from '../IsNonConventionalLoanRight';

describe('IsNonConventionalLoanRight Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<IsNonConventionalLoanRight/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
