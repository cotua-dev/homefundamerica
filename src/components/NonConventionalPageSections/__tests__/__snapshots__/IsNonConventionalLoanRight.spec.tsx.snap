// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsNonConventionalLoanRight Component renders 1`] = `
<section
  id="isNonConventionalLoanRightSection"
>
  <div>
    <h2>
      Is a Non-Conventional Loan Right For You?
    </h2>
    <strong>
      Non-conventional loans may be the right solution when looking for:
    </strong>
    <ul>
      <li>
        Expanded guidelines and enhanced options
      </li>
      <li>
        Tailored mortgage solutions for unique financial situations
      </li>
      <li>
        Customized Portfolio Loans and Interest-Only Mortgages
      </li>
    </ul>
    <button
      type="button"
    >
      Apply Now
    </button>
  </div>
  <div>
    <img
      alt="Is a non conventional loan right image"
    />
  </div>
</section>
`;
